dll/win32/shell32/shlexec.cpp

```diff
@@ -1553,20 +1553,31 @@ static HRESULT ShellExecute_ContextMenuVerb(LPSHELLEXECUTEINFOW sei)
     if (FAILED_UNEXPECTEDLY(hr))
         return hr;
 
+    BOOL fDefault = !sei->lpVerb || !sei->lpVerb[0];
```

Comment:
Could you move this one a little bit further down, just before the `if (!fDefault)` ?

Reply:
No, this fixes a bug where ici.lpVerb was tested but it is sei->lpVerb that needs to be tested (`__SHCloneStrWtoA(&verb, sei->lpVerb);` hides the true NULL when that verb buffer is assigned to ici.lpVerb)

Reply:
Ah, but I wasn't suggesting you use ici.lpVerb in the comparison. I was just suggesting moving this line verbatim just before the if (!fDefault).
The sei->lpVerb (pointer and its contents) aren't changes there, are they?

Reply:
No but someone might be tempted to use ini.lpVerb instead (again) since it looks like a valid thing to do.

Reply:
understandable